backend-ci/tests/Integration/ProductSkuCrossValidationTest.php

```
@@ -0,0 +1,113 @@
+        $p1 = $this->seedProduct(['code' => 'P001']);
+        $this->seedVariant($p1, 'SKU-111');
+
+        $this->expectException(InvalidArgumentException::class);
+        $this->expectExceptionMessage('Mã sản phẩm đã tồn tại trong danh sách phiên bản');
+
+        $this->productService->create(['code' => 'SKU-111', 'name' => 'Clash product']);
+    }
+
+    public function test_variant_sku_cannot_duplicate_product_code(): void
+    {
+        $p1 = $this->seedProduct(['code' => 'PABC']);
```

PHPMD (2.15.0)

Avoid unused local variables such as '$p1'. (undefined)

(UnusedLocalVariable)